### PR TITLE
[Merged by Bors] - Document `bevy_math`

### PR DESCRIPTION
**Pull request by KDecay**
# Objective

- Part of #3492

## Solution

- Document the `bevy_math` crate and add the `#![warn(missing_docs)]` lint.

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
Good! I'm going to call this one "trivial".

**Comment by alice-i-cecile** [comment]
Going to wait a bit to let folks double-check for nits :)

File: crates/bevy_math/src/lib.rs [hunk 1]
```diff
@@ -1,9 +1,17 @@
-pub use glam::*;
+//! Provides math types and functionality for the Bevy game engine.
+//!
+//! The commonly used types are vectors like [`Vec2`] and [`Vec3`] and
+//! matrices like [`Mat3`] and [`Mat4`].
```
Comment (devil-ira) [inline crates/bevy_math/src/lib.rs:4]: I think Quat deserves a mention here

**Review by alice-i-cecile — APPROVED** [review]

**Pull request closed without merging**